src/deploy/RPM_build/noobaa.spec

```diff
@@ -48,23 +48,21 @@ ln -s /usr/local/noobaa-core/node/bin/node $RPM_BUILD_ROOT/usr/local/noobaa-core
 ln -s /usr/local/noobaa-core/node/bin/npm $RPM_BUILD_ROOT/usr/local/noobaa-core/bin/npm
 ln -s /usr/local/noobaa-core/node/bin/npx $RPM_BUILD_ROOT/usr/local/noobaa-core/bin/npx
 
-mkdir -p $RPM_BUILD_ROOT/etc/systemd/system/
-cp -R %{_builddir}/%{name}-%{version}-%{revision}/noobaa/src/deploy/noobaa_nsfs.service $RPM_BUILD_ROOT/etc/systemd/system/noobaa_nsfs.service
+mkdir -p $RPM_BUILD_ROOT/usr/lib/systemd/system/
+cp -R %{_builddir}/%{name}-%{version}-%{revision}/noobaa/src/deploy/noobaa_nsfs.service $RPM_BUILD_ROOT/usr/lib/systemd/system/noobaa_nsfs.service
 ln -s /usr/local/noobaa-core/src/deploy/nsfs_env.env $RPM_BUILD_ROOT/usr/local/noobaa-core/nsfs_env.env
 mkdir -p $RPM_BUILD_ROOT/etc/noobaa.conf.d/
 
 mkdir -p $RPM_BUILD_ROOT/etc/rsyslog.d/
-ln -s /usr/local/noobaa-core/src/deploy/standalone/noobaa_syslog.conf $RPM_BUILD_ROOT/etc/rsyslog.d/noobaa_syslog.conf
-ln -s /usr/local/noobaa-core/src/deploy/standalone/noobaa_rsyslog.conf $RPM_BUILD_ROOT/etc/rsyslog.d/noobaa_rsyslog.conf
+cp -R $RPM_BUILD_ROOT/usr/local/noobaa-core/src/deploy/standalone/noobaa_syslog.conf $RPM_BUILD_ROOT/etc/rsyslog.d/noobaa_syslog.conf
 
 mkdir -p $RPM_BUILD_ROOT/etc/logrotate.d/noobaa
 ln -s /usr/local/noobaa-core/src/deploy/standalone/logrotate_noobaa.conf $RPM_BUILD_ROOT/etc/logrotate.d/noobaa/logrotate_noobaa.conf
 
 %files
 /usr/local/noobaa-core
-/etc/systemd/system/noobaa_nsfs.service
+/usr/lib/systemd/system/noobaa_nsfs.service
```

Comment:
@janfrode @guymguym Guy and I discussed this location replacement, and we found that Ganesha also store their service under /usr/lib/systemd/system/, @janfrode WDYT? any objections?

Reply:
Yes, my understanding is that the RPM packaging should install the default service configuration under /usr/lib/systemd/, and then that the local customizations are supposed to be done in /etc/systemd. Checking the fedora rpm packaging guidelines, it states that you should put them in %{_unitdir} macro.

https://fedoraproject.org/wiki/Packaging:Systemd

So I would suggest:

```
# diff -u noobaa.spec.old noobaa.spec
--- noobaa.spec.old     2024-02-08 10:49:51.689560030 +0100
+++ noobaa.spec 2024-02-08 10:53:45.497544084 +0100
@@ -18,6 +18,8 @@
 Source0:       %{noobaatar}
 Source1:    %{nodetar}

+BuildRequires: systemd
+
 Recommends: jemalloc

 %global __os_install_post %{nil}
@@ -62,7 +64,7 @@

 %files
 /usr/local/noobaa-core
-/etc/systemd/system/noobaa_nsfs.service
+%{_unitdir}/noobaa_nsfs.service
 /etc/logrotate.d/noobaa/logrotate_noobaa.conf
 /etc/rsyslog.d/noobaa_rsyslog.conf
 /etc/rsyslog.d/noobaa_syslog.conf
```

Reply:
thanks @janfrode 
@piste-jp-ibm WDYT?

Reply:
I have realized such kind of macro is predefined.

OK, let's move from hard coded `/usr/lib/systemd/system/` to the macro `%{_unitdir}`.

@janfrode , Good to see you again.

I would like to know the line `BuildRequires: systemd` is really required or not. In my understanding `systemd` is not required on the build phase.

Reply:
@romayalon , @janfrode ,

I updated the code.

1. Use the macro `%{_unitdir}`. I modified more few lines to use this, making directory and copying the unit file
2. Do not include `BuildRequires: systemd` because build itself is not corrupted at all

Please let me know if you have any concern.

Reply:
I added the `BuildRequires: systemd` because the Fedora packaging guideline said it was required:

https://fedoraproject.org/wiki/Packaging:Systemd

> Please note that in order for the %{_unitdir} macro to exist, your package must have: BuildRequires: systemd

I have no opinion on if that's right or not.

And very nice to see you too here Atsushi. Hope that means what I think it means :-)